### PR TITLE
Distributing software module

### PR DESCRIPTION
**Pull request by JaroCamphuijsen**
Describing various concepts regarding dependency management and isolation of the software environment:

### CODE REVIEW

**Comment by JaroCamphuijsen** [comment]
From [this blogpost](https://medium.com/kitchen-sink-data-science/software-fundamentals-for-machine-learning-series-understanding-the-why-of-vms-containers-89621cf66d23):

1. Virtual environments, Docker containers, & Virtual Machines represent increasing levels of abstraction and control.
2. By analogy, you can compare virtual environments to project folders, Docker images to file systems, and virtual machines as everything but the hardware.
3. Virtual environments are used to isolate Python project dependencies, Docker containers are used to isolate applications and their required binaries and libraries, & Virtual Machines are for managing multiple guest operating systems. These technologies aren’t mutually exclusive and can be used in tandem.

File: modules/distributing/slides-distributing.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# Distributing Software
+
+===
+
+<!-- .slide: data-state="standard" -->
+
+## Why distribute?
+
+- For your future self
+- For others that might be interesting
```
Comment (LourensVeen) [inline modules/distributing/slides-distributing.md:19]: interested

**Review by LourensVeen — APPROVED** [review]
Okay, I'd say this looks good for now. I'd like to maybe integrate the slides a bit better with the stuff I wrote, and rephrase a few things there, but for now let's go :smile:.

File: modules/distributing/distributing.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+title: Distributing Software
+type: reading
+order: 4
+---
+
+# Distributing software (10 minutes)
+
+How do you make it easy for someone else to obtain a copy and get it set up on their computer so that they can use it?
+
+Modern software contsists of an often large collection of components (libraries, packages) that are combined together to form an application. This whole collection needs to be reproduced on the computer of the user for things to work. There are two ways of doing that: 1) combining them all together on the computer of the developer, and then wrapping everything up into a package, installer, container image, or VM image that is sent to the user, or 2) putting the components that you made yourself on the Internet (as a package), and relying on the user to download the other components (packages) and assembling it all together into a working application.
```
Comment (raar1) [inline modules/distributing/distributing.md:11]:
```suggestion
Modern software consists of an often large collection of components (libraries, packages) that are combined together to form an application. This whole collection needs to be reproduced on the computer of the user for things to work. There are two ways of doing that: 1) combining them all together on the computer of the developer, and then wrapping everything up into a package, installer, container image, or VM image that is sent to the user, or 2) putting the components that you made yourself on the Internet (as a package), and relying on the user to download the other components (packages) and assembling it all together into a working application.
```

File: modules/distributing/slides-distributing.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+etc...
+
+note:
+Specify the dependencies in a file and let the user build their own environment, container or vm.
+
+===
+
+## Drawbacks
+
+Large amount of isolation enhances reproducibility but decreases flexibility.
```
Comment (raar1) [inline modules/distributing/slides-distributing.md:100]: How are you defining flexibility here? Ability to use the software as a dependency somewhere else?

**Review by raar1 — APPROVED** [review]
OK, very quick review, but I think the story seems good.

**Pull request merged**